src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -1352,27 +1351,19 @@ public static StringBuilder boundStackTrace(Throwable e, int limit, int counter,
             return out;
 
         if (!visited.add(e))
-            return out.append("[CIRCULAR REFERENCE: ").append(e.getClass().getName()).append(": ").append(e.getMessage()).append("]");
+            return out.append("[CIRCULAR REFERENCE: ").append(e.getClass().getName() + ": " + e.getMessage()).append("]").append('\n');
```

Comment:
don't do string concat, use the builder.  the compiler will replace `e.getClass().getName() + ": " + e.getMessage()` with a brand new StringBuilder, so is slightly more costly than what you had before